src/bufr/BufrParser/BufrCollectors/BufrAccumulator.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "BufrAccumulator.h"
+
+
+namespace Ingester
+{
+    BufrAccumulator::BufrAccumulator(Eigen::Index numColumns, Eigen::Index blockSize) :
+        dataArray_(blockSize, numColumns),
+        numColumns_(numColumns),
+        numDataRows_(0),
+        blockSize_(blockSize)
+    {
+    }
+
+    void BufrAccumulator::addRow(double* newRow)
+    {
+        if (numDataRows_ + 1 > dataArray_.rows())
+        {
+            dataArray_.conservativeResize(dataArray_.rows() + blockSize_, numColumns_);
+        }
+
+        dataArray_.row(numDataRows_) = Eigen::Map<IngesterArray>(newRow, 1, numColumns_);
+        numDataRows_++;
+    }
+
+    IngesterArray BufrAccumulator::getData(Eigen::Index startCol, const Channels &channels)
+    {
+        IngesterArray resultArr(numDataRows_, channels.size());
+        unsigned int colIdx = 0;
+        for (auto col : channels)
+        {
+            unsigned int offset = startCol + (col - 1);
```

Comment:
Maybe add in a check that col != 0. Sure, channels are expected to start at one, but if someone makes a mistake then you have an unbounded offset.